### PR TITLE
Test coverage for `app_extension` target types

### LINKED ISSUE #63: Build of `app_extension` target is not covered by test suite
The following changes would not be detected by the existing test suite:

1:

```diff
diff --git a/lib/pbxProject.js b/lib/pbxProject.js
index 0e0f8f9..5894d91 100644
--- a/lib/pbxProject.js
+++ b/lib/pbxProject.js
@@ -1466,19 +1466,6 @@ pbxProject.prototype.addTarget = function(name, type, subfolder) {
     // Target: Add to PBXNativeTarget section
     this.addToPbxNativeTargetSection(target)
 
-    // Product: Embed (only for "extension"-type targets)
-    if (targetType === 'app_extension') {
-
-        // Create CopyFiles phase in first target
-        this.addBuildPhase([], 'PBXCopyFilesBuildPhase', 'Copy Files', this.getFirstTarget().uuid,  targetType)
-
-        // Add product to CopyFiles phase
-        this.addToPbxCopyfilesBuildPhase(productFile)
-
-       // this.addBuildPhaseToTarget(newPhase.buildPhase, this.getFirstTarget().uuid)
-
-    };
-
     // Target: Add uuid to root project
     this.addToPbxProjectSection(target);
 
```

2:

```diff
diff --git a/lib/pbxProject.js b/lib/pbxProject.js
index 0e0f8f9..d7e65da 100644
--- a/lib/pbxProject.js
+++ b/lib/pbxProject.js
@@ -1555,7 +1555,7 @@ function pbxCopyFilesBuildPhaseObj(obj, folderType, subfolderPath, phaseName) {
      // Add additional properties for 'CopyFiles' build phase
     var DESTINATION_BY_TARGETTYPE = {
         application: 'wrapper',
-        app_extension: 'plugins',
+        app_extension: '',
         bundle: 'wrapper',
         command_line_tool: 'wrapper',
         dynamic_library: 'products_directory',
```

3:

```diff
diff --git a/lib/pbxProject.js b/lib/pbxProject.js
index 0e0f8f9..7301a16 100644
--- a/lib/pbxProject.js
+++ b/lib/pbxProject.js
@@ -1555,7 +1555,6 @@ function pbxCopyFilesBuildPhaseObj(obj, folderType, subfolderPath, phaseName) {
      // Add additional properties for 'CopyFiles' build phase
     var DESTINATION_BY_TARGETTYPE = {
         application: 'wrapper',
-        app_extension: 'plugins',
         bundle: 'wrapper',
         command_line_tool: 'wrapper',
         dynamic_library: 'products_directory',
```

I think the test coverage is needed to help ensure the functionality would not be broken by future changes.

Detected through mutation testing (PR #61) of watchkit2 updates (PR #56).

### PR DESCRIPTION
**Pull request by l3ender**
Resolves #63 by adding more extensive coverage for target types that can be added to a project.

### CODE REVIEW

File: test/addTarget.js [hunk 1]
```diff
@@ -18,6 +18,7 @@
 var fullProject = require('./fixtures/full-project')
     fullProjectStr = JSON.stringify(fullProject),
     pbx = require('../lib/pbxProject'),
+    pbxFile = require('../lib/pbxFile'),
```
Comment (brodybits) [inline test/addTarget.js]: Would you mind if we move this fix into its own PR?

File: test/addTarget.js [hunk 2]
```diff
@@ -101,5 +102,25 @@ exports.addTarget = {
...
+    },
+    'should not add build phase for non-extension target': function (test) {
+        var target = proj.addTarget(TARGET_NAME, 'application');
+        test.ok(target.uuid);
+
+        var phases = proj.pbxCopyfilesBuildPhaseObj(target.uuid);
+        test.ok(!phases);
+
+        test.done();
+    },
```
Comment (brodybits) [inline test/addTarget.js]:
```suggestion
    }
```

for the sake of consistency

File: test/addTarget.js (hunk 1 above)
Comment (l3ender) [reply test/addTarget.js]: Sure. For something like this do you prefer an issue raised, and then a PR resolving the issue? Or just a PR with a description?

Comment (brodybits) [reply test/addTarget.js]: > do you prefer an issue raised

not needed

> just a PR with a description

That would be great, thanks!

Comment (l3ender) [reply test/addTarget.js]: https://github.com/apache/cordova-node-xcode/pull/67 - thanks.
